src/plugins/data/server/ui_settings.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { schema } from '@kbn/config-schema';
+
+import { UiSettingsParams } from 'kibana/server';
+import { META_FIELDS_SETTING, DOC_HIGHLIGHT_SETTING } from '../common';
+
+export const uiSettings: Record<string, UiSettingsParams> = {
+  [META_FIELDS_SETTING]: {
+    name: i18n.translate('data.advancedSettings.metaFieldsTitle', {
+      defaultMessage: 'Meta fields',
+    }),
+    value: ['_source', '_id', '_type', '_index', '_score'],
+    description: i18n.translate('data.advancedSettings.metaFieldsText', {
+      defaultMessage:
+        'Fields that exist outside of _source to merge into our document when displaying it',
+    }),
+    schema: schema.arrayOf(schema.string()),
+  },
```

Comment:
Optional: I'm wondering whether we know all of them to declare validation as a union of well-known strings.

Reply:
here are mapping fields. but not all e.g. _score of the query context is missing, think it should stay flexible, just in case 
https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-fields.html